### ISSUE TITLE
Query Support

### ISSUE THREAD

**Issue by Alex-Muirhead** [SeniorMars/tree-sitter-typst#2]
This is more of a question than an issue, but do you intend on supporting Tree-Sitter queries in the near future? I couldn't see any mention of it in the README, however I'm aware this is a young project. 

I haven't worked on a tree-sitter grammar before, but I'd be happy to try and help if there's basic grunt work to do.

**Comment by SeniorMars** [comment]
Yes! I plan to implement all possible features within neovim especially conceal!

I'm moving at a slow pace because this language is tough to parse. It uses whitespace as a token separator SOMETIMES, so I had to rewrite everything to follow the language spec. Once I'm done with this rewrite that covers all the features I've had so far, it should be straightforward to implement the rest!

**Comment by Alex-Muirhead** [comment]
Amazing, I'm keen to have all those features! I'm using it from helix myself, so that will work great for me too. 

Argh yes, I've noticed those complications as I've scrolled through some of your notes and implementation.

**Comment by lf-** [comment]
If you run into *really* funny whitespace behaviour similar to how Python and Haskell use indents in place of braces, you might have to make like tree-sitter-haskell and write a custom lexer, although that is ... not a task for the faint of heart, as a contributor to that one.

**Comment by SeniorMars** [comment]
It's funny you say that, I'm working on the custom scanner right now and looking at python and haskell for so much reference. This language is soo big. It's like a full markdown parser + a python parser + latex math mode. It's not really fun.

**Comment by Slickytail** [comment]
I've begun writing highlight queries (for neovim, at least). It really doesn't take very long, and I expect to have highlights written for all the currently supported syntax pretty soon.

**Comment by SeniorMars** [comment]
Highlights and other queries have been low on my priority list as I thought it would be much more important to complete markup. I was super busy last week, but I also realized how to implement the rest of the language: i.e., rewrite scanner.c.


We can leave a lot of the code for code mode intact, but for other markup and math mode the best way would be to write a custom lexer :(. I have time now, so I'll be working on typst-projects again.

**Comment by jooooscha** [comment]
Any updates on this?

Any way I can help to speed things up?

**Comment by Freed-Wu** [comment]
> Query Support

https://github.com/SeniorMars/tree-sitter-typst/tree/main/queries/typst doesn't have [textobjects.scm](https://github.com/nvim-treesitter/nvim-treesitter-textobjects), which allow we `dif` to delete a function and `]m` to go to next function definition.

**Comment by SeniorMars** [comment]
@Freed-Wu Should be fixed on the latest commit. Please let me know if there are changes that need to be made!
